tests/functional/test_config_groups.py

```diff
@@ -0,0 +1,165 @@
+            test_host_1 = provider.host_create(fqdn=self.FIRST_HOST)
+        with allure.step("Create host for host candidate check"):
+            test_host_2 = provider.host_create(fqdn=self.SECOND_HOST)
+        return test_host_1, test_host_2
+
+    @allure.step('Check error')
+    def assert_that_host_add_is_unavaliable(self, error):
+        with allure.step(f"Check that error is '{GROUP_CONFIG_HOST_ERROR.code}'"):
+            GROUP_CONFIG_HOST_ERROR.equal(error)
+        with allure.step(f"Check error message is '{self.HOST_ERROR_MESSAGE}'"):
+            assert (
+                self.HOST_ERROR_MESSAGE in error.value.error['desc']
+            ), f"Should be error message '{self.HOST_ERROR_MESSAGE}'"
+
+    @allure.step("Check that the only second host is present in candidates on second provider group")
+    def assert_host_candidate_equal_expected(self, group: HostList, expected_hosts: int):
+        with allure.step(f"Check that {expected_hosts} hosts are available in group"):
+            assert len(group) == expected_hosts, f"{expected_hosts} hosts should be available in group"
+        with allure.step(f"Check that host '{self.SECOND_HOST}' is available in group"):
+            assert group[0].fqdn == self.SECOND_HOST, f"Should be available host '{self.SECOND_HOST}'"
+
+    @pytest.fixture()
+    def cluster_with_components(
+        self, create_two_hosts, cluster: Cluster, provider: Provider
+    ) -> Tuple[Service, Host, Host]:
+        """Add service, two hosts and create components to check intersection in config groups"""
+
+        service = cluster.service_add(name='test_service_1')
+        test_host_1, test_host_2 = create_two_hosts
+        cluster.host_add(test_host_1)
+        cluster.host_add(test_host_2)
+        cluster.hostcomponent_set(
+            (test_host_1, service.component(name=self.FIRST_COMPONENT_NAME)),
+            (test_host_2, service.component(name=self.FIRST_COMPONENT_NAME)),
+            (test_host_2, service.component(name=self.SECOND_COMPONENT_NAME)),
+        )
+        return service, test_host_1, test_host_2
+
+    def test_that_groups_not_allowed_to_intersect_in_cluster(self, sdk_client_fs, cluster, create_two_hosts):
+        """Test that groups are not allowed to intersect in cluster"""
+
+        test_host_1, test_host_2 = create_two_hosts
+        cluster.host_add(test_host_1)
+        cluster.host_add(test_host_2)
+        with allure.step("Create config group for cluster and add the first host"):
+            cluster_group = cluster.group_config_create(name=self.FIRST_GROUP)
+            cluster_group.host_add(test_host_1)
+        with allure.step("Create the second group for cluster and check that not allowed to add the first host to it"):
+            cluster_group_2 = cluster.group_config_create(name=self.SECOND_GROUP)
+            with pytest.raises(ErrorMessage) as e:
+                cluster_group_2.host_add(test_host_1)
```

Comment:
I think you can move this generic step to `assert_that_host_add_is_unavaliable`
```
            with pytest.raises(ErrorMessage) as e:
                cluster_group_2.host_add(test_host_1)
```

Reply:
moved